signalapp/Signal-Desktop
# (4878) Fix tooltip exceeding desired boundary

### First time contributor checklist:

- [X] I have read the [README](https://github.com/signalapp/Signal-Desktop/blob/master/README.md) and [Contributor Guidelines](https://github.com/signalapp/Signal-Desktop/blob/master/CONTRIBUTING.md)
- [X] I have signed the [Contributor Licence Agreement](https://signal.org/cla/)

### Contributor checklist:

- [X] My contribution is **not** related to translations. _Please submit translation changes via our [Signal Desktop Transifex project](https://www.transifex.com/signalapp/signal-desktop/)._
- [X] My changes are [rebased](https://medium.freecodecamp.org/git-rebase-and-the-golden-rule-explained-70715eccc372) on the latest [`development`](https://github.com/signalapp/Signal-Desktop/tree/development) branch
- [X] A `yarn ready` run passes successfully ([more about tests here](https://github.com/signalapp/Signal-Desktop/blob/master/CONTRIBUTING.md#tests))

### Description


Fixes #4878 

In some instances the popper tooltip is cut off screen due to the left pane being included in the boundary that Popper observes.

This PR adds the ability to explicitly set the boundaries element that popper uses inside `<Tooltip />`.

### Set of Changes

* add optional popperBoundariesElement prop to Tooltip 
* set popperBoundariesElement to be the title container in <ConversationHeader />

### Testing steps

Manual testing:

Platform: macOS v. 10.15.7

Steps:

- `yarn start`
- Viewed contacts with varying name lengths to check that the tooltip was moving correctly as per the <ConversationHeader /> title boundary.

### Example
![Popper Tooltip with modified boundary](https://user-images.githubusercontent.com/35961363/104970477-847a7c00-59e3-11eb-8236-e2a76e530e58.png)
